sklearn/neighbors/_base.py

```diff
@@ -619,6 +613,21 @@ def _fit(self, X, y=None):
                 else:
                     self._fit_method = "brute"
 
+        if (
+            self.effective_metric_ in ["wminkowski", "minkowski"]
+            and self.effective_metric_params_["p"] < 1
+        ):
+            # For p < 1 minkowski is not a valid metric as it does not
+            # satisfy triangular inequality, but that is workable for
+            # bruteforce algorithm. But kd_tree and ball_tree requires
+            # valid metric to work
+            if self._fit_method == "brute":
+                warnings.warn("for p < 1 minkowski is not a valid metric")
+            else:
+                raise ValueError(
+                    "p must be greater or equal to one for minkowski metric"
+                )
```

Comment:
When modifying custom inputs validation (such as here), tests must be added.

Here, tests must be added in `test_neighbors.py` to assert that all the interfaces extending `NeighborsBase` must at `fit`-time:
  - when `algo in ("brute", "auto")` and  `0 < p < 1`:
    - set `_fit_method="brute"`
    - raise a `UserWarning` indicating that Minkowski distances are not valid distance metrics for 0 < p < 1
  - when `algo in ("kd_tree", "ball_tree")` and `0 < p < 1`:
    - raise a `ValueError` indicating that:
        - `algo="kd_tree"` and `algo="ball_tree"` are both unusable for this configuration because Minkowski distances are not valid distance metrics for `0 < p < 1`
        - `p >= 1` or `algo="brute"` must be set.

To better have a seen on how to perform this, you can have a look at some others tests in this file  this, such as this one:

https://github.com/scikit-learn/scikit-learn/blob/5b45d1fa7faad53db1a0fac9b15d23cb22375031/sklearn/neighbors/tests/test_neighbors.py#L1455-L1498

Reply:
Thanks @jjerphan, I'm modifying tests accordingly.😀
Also, I've not changed the logic for when `algorithm="auto"`, from the first bullet point in your comment, it seems that if `algorithm` is set to `"auto"`  and `0 < p < 1`, then irrespective of other things, `_fit_method` should be set to `brute`.   

So should I change the implementation for doing the same or current is fine?🤔

Reply:
@jjerphan I've added tests as you described.  Based on your first testcase, I've also made a little tweak in logic for `auto` algorithm, now if `algorithm="auto"` and `p < 1` then `_fit_method` will be set to `brute`.  

Let me know what further changes are necessary.